src/v/config/configuration.cc

```diff
@@ -1628,7 +1630,22 @@ configuration::configuration()
       {.needs_restart = needs_restart::no,
        .example = "2147483648000",
        .visibility = visibility::user},
-      std::nullopt)
+      std::nullopt,
+      property<std::optional<size_t>>::noop_validator,
+      legacy_default<std::optional<size_t>>(std::nullopt, legacy_version{9}))
+  , retention_local_target_capacity_percent(
+      *this,
+      "retention_local_target_capacity_percent",
+      "The target capacity in percent of unreserved space that log storage "
+      "will try to use before additional retention rules will take over to "
+      "trim data in order to meet the target. When no target is specified "
+      "storage usage is unbounded.",
+      {.needs_restart = needs_restart::no,
+       .example = "70.0",
+       .visibility = visibility::user},
+      80.0,
```

Comment:
Shouldn't `retention_local_target_capacity_percent`, `disk_reservation_percent` and `cloud_storage_cache_size_percent` add up to 100 (for the case when the cloud storage cache uses the same disk)

Reply:
I think we want to keep some free space. Flash translation layer of the SSD drive needs some space to be happy. Otherwise the latency may spike because of GC.

Reply:
Yup, I agree. Isn't `disk_reservation_percent` meant to be the slice of disk that we keep free though?